src/components/MasterItems.js

```diff
@@ -49,11 +42,22 @@ class MasterItems extends Component {
     this.setState({ entries: [] });
   }
 
+  addComment(entryIndex, e) {
+    const updatedEntry = this.state.entries[entryIndex];
+    updatedEntry.comment = e.target.value;
```

Comment:
I think with const you can't change the reference of the variable, so can't reassign.